app/models/concerns/client_image_consumer.rb

```diff
@@ -14,7 +14,7 @@ module ClientImageConsumer
     def image(_cache_for = 10.minutes)
       # Check first for locally uploaded image or a cached ETO image
       # Otherwise, check for an image connected to the source client
-      local_client_image_data || source_clients.detect(&:image_for_source_client)&.image_for_source_client
+      local_client_image_data.presence || source_clients.detect(&:image_for_source_client)&.image_for_source_client.presence
     end
 
     def image_for_source_client(_cache_for = 10.minutes)
```

Comment:
I'm not sure why this method returns an empty string when there is no image, but it seems like that's a holdover from `stable`.

Here is the code from stable, which explicitly returns `''` in some cases and `nil` in others. @eanders Do you know why that is?

https://github.com/greenriver/hmis-warehouse/blob/6702d55a7a29cba68b29bbf0342cadd83de7754a/app/models/grda_warehouse/hud/client.rb#L1484-L1505

Reply:
Yeah, thanks, we were continuing to return an empty string since that's what it used to do, but we didn't investigate further to determine if that was the correct behavior!  We should probably convert them all to nils, but what you have should work for now.  Thanks!

Reply:
What I had actually didn't work, calling `.presence` on the image string will fail with `ArgumentError: invalid byte sequence in UTF-8`. You can call `blank?` or `empty?` but not `present?`... the `client_visibility_config_variations_legacy_spec` helped me figure that out. I will push another change

Reply:
good catch!

I wonder if we should try and set the encoding correctly on these data strings to avoid this type of exception. For example `image_for_source_client` should call something like `data.force_encoding(Encoding::BINARY))` before returning